cmd/controller/main.go

```diff
@@ -239,6 +239,12 @@ func main() {
 		rs = append(rs, gasController)
 	}
 
+	go func(r runner) {
```

Comment:
Three things:
1. I would just inline this as:
```
go func() {
  if err := server.Run(ctx, [...]) [...]
}()
```

OR, you can make this a helper function and use it here and below, like:

```
runRunner := func(r runner) {
}
go runRunner(server)
```
(then later use the helper again)

2. Comment here why we need to start `server` early.

Reply:
turned it into a helper function